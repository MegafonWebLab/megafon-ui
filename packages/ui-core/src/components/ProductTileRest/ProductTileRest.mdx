---
name: ProductTileRest
menu: Components
---

import { Playground, PropsTable } from 'docz';
import ProductTileRest from './ProductTileRest';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';

export const Wrapper = ({children}) => <div style={{ padding: '20px', display: 'flex' }}>{children}</div>

# ProductTile

## Properties ProductTileRest
<PropsTable of={ProductTileRest} />

## Basic usage

export const tariff = {
    link: '/tariffs/vklyuchaysya/smotri.html',
    linkTarget: '_self',
    moreLinkText: 'Подробнее о тарифе',
    showMoreLink: true,
    buyLink: '/zakaz/?tariff=look',
    buyLinkTarget: '_self',
    buyButtonText: 'Выбрать',
    showBuyButton: true,
    connectLink: '#',
    connectLinkTarget: '_self',
    connectButtonText: 'Перейти',
    showConnectButton: true,
    "title": "Включайся! Смотри",
    "description": "Описание",
    "payment": {
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": ''
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут",
            "title": 'Безлим',
            isDelim: true,
        },
        {
            "value": 100,
            "unit": "смс",
            "title": 'Безлим',
            isDelim: false,
        },
        {
            "value": 3,
            "unit": "ГБ",
            "title": 'Безлим',
            isDelim: false,
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};

export const tariff2 = {
    link: '/tariffs/vklyuchaysya/smotri.html',
    buyLink: '/zakaz/?tariff=look',
    connectLink: '#',
    shopTag: 'tag',
    "title": "Включайся! Смотри",
    "description": "Описание",
    "payment": {
        title: 'При покупке новой SIM–карты',
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": '600'
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут"
        },
        {
            "value": 100,
            "unit": "смс"
        },
        {
            "value": 3,
            "unit": "ГБ"
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "Параметры",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};

<Playground wrapper={Wrapper}>
    <div style={{ width: '300px', margin: '0 20px 0 0' }}>
        <ProductTileRest {...tariff} />
    </div>
    <div style={{ width: '300px' }}>
        <ProductTileRest {...tariff2} />
    </div>
</Playground>

## Included props

``` javascript
export const tariff = {
    buyButtonText: 'Купить',
    connectButtonText: 'Перейти на тариф',
    "title": "Включайся! Смотри",
    "description": "Описание",
    "link": "/tariffs/vklyuchaysya/smotri.html",
    "buyLink": "/zakaz/?tariff=look",
    "payment": {
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": ''
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут"
        },
        {
            "value": 100,
            "unit": "смс"
        },
        {
            "value": 3,
            "unit": "ГБ"
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};

export const tariff2 = {
    buyButtonText: 'Купить',
    connectButtonText: 'Перейти на тариф',
    "title": "Включайся! Смотри",
    "description": "Описание",
    "link": "/tariffs/vklyuchaysya/smotri.html",
    "buyLink": "/zakaz/?tariff=look",
    "payment": {
        title: 'При покупке новой SIM–карты',
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": '600'
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут"
        },
        {
            "value": 100,
            "unit": "смс"
        },
        {
            "value": 3,
            "unit": "ГБ"
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};
```
