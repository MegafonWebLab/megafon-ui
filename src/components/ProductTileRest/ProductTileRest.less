@import '~styles/base.less';

@b: e('.mfui-product-tile-rest');

@{b} {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    padding: 24px 8px 16px;
    border-radius: 9px;
    background-color: @clearWhite;
    box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
    box-sizing: border-box;

    &:hover {
        box-shadow: 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }

    @media @desktopMU {
        padding: 32px 16px 16px;
    }

    &__info,
    &__buy {
        text-align: center;
    }

    &__header {
        font-size: 18px;
        font-weight: 500;
        margin-bottom: 4px;
        letter-spacing: 0.5px;

        @media @desktopMU {
            font-size: 20px;
            line-height: 28px;
            margin-bottom: 0;
        }
    }

    &__detail-link {
        margin-bottom: 14px;
        display: block;
    }

    &__options {
        margin-bottom: 12px;

        @media @mobileSM {
            margin-bottom: 24px;
        }

        @media @desktopMU {
            margin-bottom: 24px;
        }
    }

    &__option {
        &:not(:last-child) {
            margin-bottom: 8px;
        }
    }

    &__option-title {
        line-height: 24px;
        font-weight: 500;
        font-size: 15px;

        a {
            .default-link();
            word-break: break-word;
        }
    }

    &__option-description {
        font-size: 12px;
        line-height: 16px;
    }

    &__description {
        font-size: 12px;
        line-height: 16px;
        margin-bottom: 25px;
    }

    &__pack {
        margin-bottom: 8px;
        font-size: 15px;
        font-weight: 500;
    }

    &__messangers {
        margin-bottom: 24px;
    }

    &__messangers-list {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;

        svg,
        img
        {
            width: 32px;
            height: 32px;
        }

        svg {
            pointer-events: none;
        }
    }

    &__messangers-description {
        font-size: 12px;
        letter-spacing: 0.33px;
        line-height: 16px;
    }

    &__reward-icon {
        svg,
        img {
            width: 32px;
            height: 32px;
        }

        svg {
            pointer-events: none;
        }
    }

    &__buy-button {
        margin-bottom: 11px;
        width: 159px;

        &_additional-margin {
            margin-bottom: 48px;

            @media @mobileSM {
                margin-bottom: 11px;
            }
        }
    }

    &__actual-price {
        font-size: 20px;
        font-weight: 500;
        line-height: 28px;
        letter-spacing: 0.5px;
        margin-bottom: 26px;

        @media @mobileSM {
            margin-bottom: 18px;
        }

        @media @desktopMU {
            margin-bottom: 24px;
        }
    }

    &__payment-period {
        font-weight: normal;
        font-size: 15px;
        line-height: 24px;
        letter-spacing: 0;
    }

    &__options-list {
        align-items: center !important;
    }

    &__discount-condition {
        font-size: 12px;
        font-weight: normal;
        line-height: 16px;
        min-height: 16px;
    }

    &__old-price-wrapper {
        display: flex;
        justify-content: center;
        flex-grow: 0;
        flex-shrink: 0;
    }
    &__old-price {
        font-size: 15px;
        position: relative;
        line-height: 20px;
        letter-spacing: 0.5px;
        color: @black40;

        &:before {
            content: '';
            position: absolute;
            width: 100%;
            height: 2px;
            margin: auto;
            top: 0;
            bottom: 0;
            background-color: @fury;
        }
    }

    &__price &__old-price-wrapper {
        visibility: hidden;
    }

    &__price_discount &__old-price-wrapper {
        visibility: visible;
    }

    &_active {
        box-shadow: inset 0 0 0 2px @greenActive, 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }

    &_active:hover {
        box-shadow: inset 0 0 0 2px @greenActive, 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }
}
