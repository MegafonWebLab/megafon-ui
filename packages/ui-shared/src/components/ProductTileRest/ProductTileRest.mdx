---
name: ProductTileRest
menu: Компоненты
route: components/product-tile-rest
---

import { Playground, Props } from 'docz';
import ProductTileRest from './ProductTileRest';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';
import { ProductTileRestWrapper, tariff, tariff2 } from './ProductTileRest.docz';

# ProductTileRest

## Properties ProductTileRest
<Props of={ProductTileRest} />

## Basic usage

<Playground>
    <ProductTileRestWrapper>
        <div style={{ width: '300px', margin: '0 20px 0 0' }}>
            <ProductTileRest {...tariff} />
        </div>
        <div style={{ width: '300px' }}>
            <ProductTileRest {...tariff2} />
        </div>
    </ProductTileRestWrapper>
</Playground>

## Included props

``` javascript
export const tariff = {
    buyButtonText: 'Купить',
    connectButtonText: 'Перейти на тариф',
    "title": "Включайся! Смотри",
    "description": "Описание",
    "link": "/tariffs/vklyuchaysya/smotri.html",
    "buyLink": "/zakaz/?tariff=look",
    "payment": {
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": ''
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут"
        },
        {
            "value": 100,
            "unit": "смс"
        },
        {
            "value": 3,
            "unit": "ГБ"
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};

export const tariff2 = {
    buyButtonText: 'Купить',
    connectButtonText: 'Перейти на тариф',
    "title": "Включайся! Смотри",
    "description": "Описание",
    "link": "/tariffs/vklyuchaysya/smotri.html",
    "buyLink": "/zakaz/?tariff=look",
    "payment": {
        title: 'При покупке новой SIM–карты',
        "value": '1000',
        "unitExtra": "за 30 дней",
        "unitValue": "₽",
        "unit": "₽ за 30 дней",
        "discount": '600'
    },
    "packs": [
        {
            "value": 1500,
            "unit": "минут"
        },
        {
            "value": 100,
            "unit": "смс"
        },
        {
            "value": 3,
            "unit": "ГБ"
        }
    ],
    "firstParams": {
        "title": "Параметры",
        "caption": "",
        "items": [
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "МегаФон ТВ",
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
            {
                "title": "VIP-тариф",
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <MegafonTv />
            },
        ]
    },
    "secondParams": [{
        "title": "Опция",
        "value": '300',
        "unit": "минут"
    }],
    "cashback": {
        "title": "Кэшбэк",
        "value": 20,
        "unit": "%"
    },
    "info": {
        "billingIds": ['sdf', 'sdf'],
        "sms": "05007897",
        "smsText": "ДА",
        "smsSite": null,
        "ivr": "05007897",
        "ussd": "<span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">789</b><span class=\"c-number-call\"><b class=\"asterisk\">*</b><b class=\"num\">7</b><b class=\"dash\">#</b><b class=\"pipe\"></b></span>",
        "lkLink": "https://lk.megafon.ru/tariffs/",
        "title": "Включайся! Смотри",
        "id": 1598
    }
};
```
