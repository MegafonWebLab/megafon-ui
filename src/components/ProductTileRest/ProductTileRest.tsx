import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './ProductTileRest.less';
import Header from '../Header/Header';
import TextLink from '../TextLink/TextLink';
import Button from '../Button/Button';
import DropdownSocialList from '../DropdownSocialList/DropdownSocialList';

interface IProductTileRestProps {
    title: string;
    description: string;
    shopTag: string;

    link: string;
    moreLinkText: string;
    showMoreLink: boolean;

    buyLink: string;
    buyButtonText: string;
    showBuyButton: boolean;

    connectLink: string;
    connectButtonText: string;
    showConnectButton: boolean;
    isBuyable: boolean;

    payment: any;
    packs: any;
    firstParams: any;
    secondParams: any;
    info: any;
    onClickConnect: any;
    onClickBuy: any;
    onClickMore: any;
}

const cn = cnCreate('mfui-product-tile-rest');
class ProductTileRest extends React.Component<IProductTileRestProps> {
    static propTypes = {
        title: PropTypes.string,
        description: PropTypes.string,
        shopTag: PropTypes.string,
        link: PropTypes.string,
        moreLinkText: PropTypes.string,
        showMoreLink: PropTypes.bool,
        isBuyable: PropTypes.bool,
        buyLink: PropTypes.string,
        buyButtonText: PropTypes.string,
        showBuyButton: PropTypes.bool,
        connectLink: PropTypes.string,
        connectButtonText: PropTypes.string,
        showConnectButton: PropTypes.bool,
        payment: PropTypes.shape({
            value: PropTypes.string.isRequired,
            unitExtra: PropTypes.string,
            unitValue: PropTypes.string,
            discount: PropTypes.string,
        }),
        packs: PropTypes.arrayOf(PropTypes.shape({
            value: PropTypes.number,
            unit: PropTypes.string,
            title: PropTypes.string,
            isDelim: PropTypes.bool,
        })),
        firstParams: PropTypes.shape({
            title: PropTypes.string,
            caption: PropTypes.string,
            icons: PropTypes.arrayOf(PropTypes.shape({
                title: PropTypes.string,
                svgIcon: PropTypes.element,
            })),
        }),
        secondParams: PropTypes.arrayOf(PropTypes.shape({
            title: PropTypes.string,
            value: PropTypes.string,
            unit: PropTypes.string,
        })),
        info: PropTypes.object,
        onClickConnect: PropTypes.func,
        onClickBuy: PropTypes.func,
        onClickMore: PropTypes.func,
    };

    static defaultProps = {
        moreLinkText: 'Подробнее',
        showMoreLink: true,
        buyButtonText: 'Купить',
        showBuyButton: true,
        connectButtonText: 'Перейти на тариф',
        showConnectButton: true,
        shopTag: '',
    };

    handleClickConnect = (e: React.SyntheticEvent<EventTarget>) => {
        const { info, shopTag, onClickConnect } = this.props;

        onClickConnect && onClickConnect({ ...info, shopTag }, e);
    }

    handleClickBuy = (e: React.SyntheticEvent<EventTarget>) => {
        const { info, shopTag, onClickBuy } = this.props;

        onClickBuy && onClickBuy({ ...info, shopTag }, e);
    }

    handleClickMore = (e: React.SyntheticEvent<EventTarget>) => {
        const { info, shopTag, onClickMore } = this.props;

        onClickMore && onClickMore({ ...info, shopTag }, e);
    }

    renderIcons() {
        const { firstParams: { title, caption, items } } = this.props;

        return (
            <div className={cn('messangers')}>
                {!!items.length &&
                    <div className={cn('messangers-list')}>
                        <DropdownSocialList
                            icons={items}
                            maxIconNumber={4}
                            className={cn('options-list')}
                        />
                    </div>
                }
                {title &&
                    <div className={cn('messangers-description')}>
                        {title}<br /> {caption}
                    </div>
                }
            </div>
        );
    }

    renderOptions() {
        const { secondParams } = this.props;

        if (!secondParams.length) {
            return null;
        }

        return (
            <div className={cn('options')}>
                {secondParams.map((param, index) => {
                    const { title, value, unit } = param;

                    return (
                        <div className={cn('option')} key={title + index}>
                            <div className={cn('option-title')}>{title}</div>
                            <div className={cn('option-description')}>{`${value} ${unit}`}</div>
                        </div>
                    );
                })}
            </div>
        );
    }

    renderShowcase() {
        const {
            payment: { value, unitExtra, unitValue, discount },
            packs,
        } =  this.props;

        return (
            <React.Fragment>
                <div className={cn('price', { discount: !!discount })}>
                    <div className={cn('discount-condition')}>При покупке новой SIM–карты</div>
                    <div className={cn('old-price-wrapper')}>
                        <div className={cn('old-price')}>{`${value} ${unitValue}`}</div>
                    </div>
                    <div className={cn('actual-price')}>{`${discount || value} ${unitValue} ${unitExtra}`}</div>
                </div>
                {packs.map((param, index) => {
                    const {
                        value: paramValue,
                        unit: paramUnit,
                        title: paramTitle,
                        isDelim,
                    } = param;

                    return (
                        <div
                            className={cn('pack')}
                            key={paramValue + paramUnit + index}
                        >
                            {isDelim ? paramTitle : `${paramValue} ${paramUnit}`}
                        </div>
                    );
                })}
            </React.Fragment>
        );
    }

    render() {
        const {
            title,
            description,
            link,
            showMoreLink,
            moreLinkText,
            connectLink,
            connectButtonText,
            showConnectButton,
            buyLink,
            showBuyButton,
            buyButtonText,
            isBuyable,
        } = this.props;
        const buyButtonColor = isBuyable ? 'green' : 'transparent-green';

        return (
            <div className={cn('')}>
                <div className={cn('info')}>
                    <Header className={cn('header')} as="h3">{title}</Header>
                    {showMoreLink &&
                        <TextLink
                            className={cn('detail-link')}
                            href={link}
                            target="_blank"
                            onClick={this.handleClickMore}
                        >
                            {moreLinkText}
                        </TextLink>
                    }
                    {this.renderShowcase()}
                    {this.renderIcons()}
                    {this.renderOptions()}
                    {description && <div className={cn('description')}>{description}</div>}
                </div>
                <div className={cn('buy')}>
                    {showBuyButton &&
                        <Button
                            className={cn('buy-button')}
                            passiveColor={buyButtonColor}
                            hoverColor="green"
                            sizeAll="medium"
                            href={buyLink}
                            onClick={this.handleClickBuy}
                        >
                            {buyButtonText}
                        </Button>
                    }
                    {showConnectButton &&
                        <TextLink
                            className={cn('detail-link')}
                            href={connectLink}
                            onClick={this.handleClickConnect}
                            target="_blank"
                        >
                            {connectButtonText}
                        </TextLink>
                    }
                </div>
            </div>
        );
    }
}

export default ProductTileRest;
