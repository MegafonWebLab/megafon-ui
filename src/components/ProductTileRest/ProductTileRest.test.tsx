import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileRest from './ProductTileRest';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';
import { IProductTileRestProps } from './ProductTileRest';

const tariff: Partial<IProductTileRestProps> = {
    showMoreLink: false,
    buyButtonText: 'Купить',
    showBuyButton: true,
    connectLink: '123123',
    connectButtonText: 'Подключиться',
    showConnectButton: false,
    title: 'Включайся! Смотри',
    'moreLinkText': 'more link',
    'description': 'asdasdasd',
    'shopTag': '123',
    'link': '/tariffs/vklyuchaysya/smotri.html',
    'buyLink': '/zakaz/?tariff=look',
    'payment': {
        'value': '1000',
        'unitExtra': 'за 30 дней',
        'unitValue': '₽',
        'unit': '₽ за 30 дней',
        'discount': '0',
    },
    'packs': [
        {
            'value': 1500,
            'unit': 'минут',
            'title': 'title',
            'isDelim': true,
        },
        {
            'value': 100,
            'unit': 'смс',
            'title': 'title',
            'isDelim': true,
        },
        {
            'value': 3,
            'unit': 'ГБ',
            'title': 'title',
            'isDelim': false,
        },
    ],
    'firstParams': {
        'title': 'Параметры',
        'caption': '',
        'items': [
            {
                'title': 'VIP-тариф',
                'caption': 'Привилегии обслуживания и другое',
                'svgIcon': <MegafonTv />,
            },
            {
                'title': 'МегаФон ТВ',
                'caption': 'Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус',
                'svgIcon': <MegafonTv />,
            },
        ],
    },
    'secondParams': [
        {
            'title': 'Кэшбэк 20%',
            'footnote': 'Участвуйте в программе и тратьте кэшбэк',
            'value': '0',
            'unit': '',
            'svgIcon': <MegafonTv />,
        },
        {
            'title': 'Безлимитные SMS',
            'footnote': '',
            'value': '55',
            'unit': '₽ за 30 дней',
            'svgIcon': <MegafonTv />,
        },
    ],
    'info': {
        'billingIds': [],
        'sms': '05007897',
        'smsText': 'ДА',
        'smsSite': null,
        'ivr': '05007897',
        'ussd': '<span class=\'c-number-call\'>sdfsdf</span>',
        'lkLink': 'https://lk.megafon.ru/tariffs/',
        'title': 'Включайся! Смотри',
        'id': 1598,
    },
};

describe('<Product >', () => {
    it('should render', () => {
        const wrapper = shallow(
            <ProductTileRest {...tariff}/>
            );

        expect(wrapper).toMatchSnapshot();
    });
});
