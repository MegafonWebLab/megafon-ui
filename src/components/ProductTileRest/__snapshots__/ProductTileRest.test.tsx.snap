// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product > should render 1`] = `
<div
  className="mfui-product-tile-rest"
>
  <div
    className="mfui-product-tile-rest__info"
  >
    <Header
      as="h3"
      className="mfui-product-tile-rest__header"
      color="black"
    >
      Включайся! Смотри
    </Header>
    <div
      className="mfui-product-tile-rest__price mfui-product-tile-rest__price_discount"
    >
      <div
        className="mfui-product-tile-rest__discount-condition"
      />
      <div
        className="mfui-product-tile-rest__old-price-wrapper"
      >
        <div
          className="mfui-product-tile-rest__old-price"
        >
          1000 ₽
        </div>
      </div>
      <div
        className="mfui-product-tile-rest__actual-price"
      >
        0 ₽ за 30 дней
      </div>
    </div>
    <div
      className="mfui-product-tile-rest__pack"
      key="1500минут0"
    >
      title
    </div>
    <div
      className="mfui-product-tile-rest__pack"
      key="100смс1"
    >
      title
    </div>
    <div
      className="mfui-product-tile-rest__pack"
      key="3ГБ2"
    >
      3 ГБ
    </div>
    <div
      className="mfui-product-tile-rest__messangers"
    >
      <div
        className="mfui-product-tile-rest__messangers-list"
      >
        <DropdownSocialList
          className="mfui-product-tile-rest__options-list"
          icons={
            Array [
              Object {
                "caption": "Привилегии обслуживания и другое",
                "svgIcon": <test-file-stub />,
                "title": "VIP-тариф",
              },
              Object {
                "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
                "svgIcon": <test-file-stub />,
                "title": "МегаФон ТВ",
              },
            ]
          }
          maxIconNumber={4}
        />
      </div>
      <div
        className="mfui-product-tile-rest__messangers-description"
      >
        Параметры
        <br />
         
      </div>
    </div>
    <div
      className="mfui-product-tile-rest__options"
    >
      <div
        className="mfui-product-tile-rest__option"
        key="Кэшбэк 20%0"
      >
        <div
          className="mfui-product-tile-rest__option-title"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Кэшбэк 20%",
            }
          }
        />
        <div
          className="mfui-product-tile-rest__option-description"
        >
          0 
        </div>
      </div>
      <div
        className="mfui-product-tile-rest__option"
        key="Безлимитные SMS1"
      >
        <div
          className="mfui-product-tile-rest__option-title"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Безлимитные SMS",
            }
          }
        />
        <div
          className="mfui-product-tile-rest__option-description"
        >
          55 ₽ за 30 дней
        </div>
      </div>
    </div>
    <div
      className="mfui-product-tile-rest__description"
    >
      asdasdasd
    </div>
  </div>
  <div
    className="mfui-product-tile-rest__buy"
  >
    <Button
      className="mfui-product-tile-rest__buy-button mfui-product-tile-rest__buy-button_additional-margin"
      disabledColor="gray"
      downColor="dark"
      hoverColor="green"
      href="/zakaz/?tariff=look"
      onClick={[Function]}
      padding={true}
      passiveColor="green"
      showSpinner={false}
      sizeAll="medium"
      type="button"
      width="auto"
    >
      Купить
    </Button>
  </div>
</div>
`;
